src/DiagnosticMessages.ts

```diff
@@ -738,85 +891,99 @@ export let DiagnosticMessages = {
         message: `Member '${memberName}' is ${accessModifierNameFromFlag(accessModifierFlag)}${accessModifierAdditionalInfo(accessModifierFlag, definingClassName)}`, // TODO: Add scopes where it was defined
-        code: 1146,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1146,
+        severity: DiagnosticSeverity.Error,
+        code: 'member-accessibility-mismatch'
     }),
     typecastStatementMustBeDeclaredAtStart: () => ({
         message: `'typecast' statement must be declared at the top of the file or beginning of function or namespace`,
-        code: 1147,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1147,
+        severity: DiagnosticSeverity.Error,
+        code: 'typecast-must-be-at-start'
     }),
     invalidTypecastStatementApplication: (foundApplication: string) => ({
         message: `'typecast' statement can only be applied to 'm', but was applied to '${foundApplication}'`,
-        code: 1148,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1148,
+        severity: DiagnosticSeverity.Error,
+        code: 'invalid-typecast-statement'
     }),
     itemCannotBeUsedAsType: (typeText: string) => ({
         message: `'${typeText}' cannot be used as a type`,
-        code: 1149,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1149,
+        severity: DiagnosticSeverity.Error,
+        code: 'item-cannot-be-used-as-type'
     }),
     expectedNewlineInConditionalCompile: () => ({
         message: `Expected newline in conditional compilation statement`,
-        code: 1151,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1151,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-newline-in-conditional-compile'
     }),
-    expectedTerminatorOnConditionalCompileBlock: () => ({
+    __unused22: () => ({
         message: `Expected '#end if', '#else if', or '#else' to terminate conditional compilation block`,
-        code: 1152,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1152,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-terminator-on-conditional-compile'
     }),
     unsafeUnmatchedTerminatorInConditionalCompileBlock: (terminator: string) => ({
         message: `Unsafe unmatched terminator '${terminator}' in conditional compilation block`,
-        code: 1153,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1153,
+        severity: DiagnosticSeverity.Error,
+        code: 'unmatched-terminator-on-conditional-compile'
     }),
     cannotFindTypeInCommentDoc: (name: string) => ({
         message: `Cannot find type '${name}' in doc comment`,
-        code: 1154,
+        legacyCode: 1154,
         data: {
             name: name
         },
-        severity: DiagnosticSeverity.Error
+        severity: DiagnosticSeverity.Error,
+        code: 'missing-type-in-doc-comment'
```

Comment:
Can we just merge this in with the other "cannot find type"? Not sure we need to clarify "in doc comment" as we're going to highlight that spot anyway...